app/frontend/src/components/designer/FormViewerMultiUpload.vue

```diff
@@ -0,0 +1,535 @@
+        JSON_OBJECT_NOT_ARRAY: 'Wrong json file format',
+        JSON_ARRAY_EMPTY: 'This json file is empty.',
+        ERROR_WHILE_VALIDATE: 'There is something wrong with this file',
+        ERROR_WHILE_CHECKVALIDITY: 'There is something wrong with this file',
+        ERROR_AFTER_VALIDATE:
+          'Some errors found, see below for more information.',
+      },
+    };
+  },
+  computed: {
+    txt_color() {
+      if (!this.error) return 'success-text';
+      else return 'fail-text';
+    },
+    fileSize() {
+      if (this.file.size < 1024) return this.file.size.toFixed(2) + ' bytes';
+      if (this.file.size < 1024 * 1024)
+        return (this.file.size / 1024).toFixed(2) + ' KB';
+      return (this.file.size / (1024 * 1024)).toFixed(2) + ' MB';
+    },
+  },
+  methods: {
+    ...mapActions('notifications', ['addNotification']),
+    addFile(e, type) {
+      if (this.block) {
+        return;
+      }
+
+      if (this.file != undefined) {
+        this.addNotification({
+          message: this.ERROR.UPLOAD_MULTIPLE_FILE_ERROR,
+          consoleError: this.ERROR.UPLOAD_MULTIPLE_FILE_ERROR,
+        });
+        return;
+      }
+      try {
+        let droppedFiles = type == 0 ? e.dataTransfer.files : [e];
+
+        if (!droppedFiles || droppedFiles == undefined) return;
+
+        if (droppedFiles.length > 1) {
+          this.addNotification({
+            message: this.ERROR.DRAG_MULPLE_FILE_ERROR,
+            consoleError: this.ERROR.DRAG_MULPLE_FILE_ERROR,
+          });
+          return;
+        }
+
+        if (droppedFiles[0]['type'] != 'application/json') {
+          this.addNotification({
+            message: this.ERROR.FILE_FORMAT_ERROR,
+            consoleError: this.ERROR.FILE_FORMAT_ERROR,
+          });
+          return;
+        }
+        let size = droppedFiles[0].size / (1024 * 1024);
+        if (size > this.max_file_size) {
+          this.addNotification({
+            message: this.ERROR.FILE_SIZE_ERROR,
+            consoleError: this.ERROR.FILE_SIZE_ERROR,
+          });
+          return;
+        }
+        this.file = droppedFiles[0];
+        this.parseFile();
+      } catch (error) {
+        this.addNotification({
+          message: this.ERROR.DRAG_MULPLE_FILE_ERROR,
+          consoleError: `${this.ERROR.DRAG_MULPLE_FILE_ERROR} ${error}`,
+        });
+        return;
+      }
+    },
+    handleFile() {
+      if (this.file == undefined) {
+        this.$refs.file.$refs.input.click();
+        // this.$refs.file.click();
+      }
+    },
+    removeFile(file) {
+      this.files = this.files.filter((f) => {
+        return f != file;
+      });
+    },
+    parseFile() {
+      try {
+        let reader = new FileReader();
+        reader.onload = function (e) {
+          this.Json = JSON.parse(e.target.result);
+        }.bind(this);
+
+        reader.onloadend = () => {
+          this.preValidateSubmission();
+        };
+        reader.readAsText(this.file);
```

Comment:
this line should come after prevalidateSubmission. onloadend is a promise, so it'll execute preValidateSubmission but can also execute this line at the same time before preValidateSubmission is complete.

Reply:
this line is entrypoint  imperatively  it gonna run before onloadend cause onloadend is a callback its call after finishing to read the file .